File: tests/fixtures/ts/rebinding/src/read_before_rebind.ts
import Stripe from 'stripe';

const stripe = new Stripe(process.env.STRIPE_KEY!);

export async function indexed(rows: any[]) {
  const charge = await stripe.charges.create({ amount: 100, currency: 'usd' });

  // Both reads in `summarise` throw. `const charge` below puts the whole body in that
  // binding's temporal dead zone, so the line above it reaches no outer object.
  function summarise() {
    const early = charge.leaked;
    const charge = rows[0];
    return [early, charge.also_leaked];
  }

  return { summarise, state: charge.status };
}

File: tests/fixtures/py/rebinding/src/read_before_rebind.py
from stripe import StripeClient

client = StripeClient("sk_test")


def indexed(rows):
    charge = client.charges.create(amount=100, currency="usd")

    # Both reads in `summarise` raise UnboundLocalError: assigning `charge` anywhere in the
    # body makes it local to the whole body, including the line above the assignment.
    def summarise():
        early = charge.leaked
        charge = rows[0]
        return early, charge.also_leaked

    return summarise, charge.status
